src/components/AnnotationsCarousel/AnnotationButton.tsx

```diff
@@ -0,0 +1,186 @@
+    return LsCommentRed;
+  } else if (ent.comment_count > 0) {
+    return LsComment;
+  }
+
+  return null;
+};
+
+export const AnnotationButton = observer(({ entity, capabilities, annotationStore, onAnnotationChange }: AnnotationButtonInterface) => {
+  const iconSize = 37;
+  const isPrediction = entity.type === 'prediction';
+  const username = userDisplayName(entity.user ?? {
+    firstName: entity.createdBy || 'Admin',
+  });
+  const [isGroundTruth, setIsGroundTruth] = useState<boolean>();
+  const [isContextMenuOpen, setIsContextMenuOpen] = useState<boolean>(false);
+
+  const CommentIcon = renderCommentIcon(entity);
+
+  useEffect(() => {
+    setIsGroundTruth(entity.ground_truth);
+  }, [entity, entity.ground_truth]);
+
+  const clickHandler = useCallback(() => {
+    const { selected, id, type } = entity;
+    
+    if (!selected) {
+      if (type === 'prediction') {
+        annotationStore.selectPrediction(id);
+      } else {
+        annotationStore.selectAnnotation(id);
+      }
+    }
+  }, [entity]);
+  const ContextMenu = ({ entity, capabilities }: AnnotationButtonInterface) => {
+    const dropdown = useDropdown();
+    const clickHandler = () => {
+      onAnnotationChange?.();
+      dropdown?.close();
+    };
+    const setGroundTruth = useCallback(() => {
+      entity.setGroundTruth(!isGroundTruth);
+      clickHandler();
+    }, [entity]);
+    const duplicateAnnotation = useCallback(() => {
+      const c = annotationStore.addAnnotationFromPrediction(entity);
+
+      window.setTimeout(() => {
+        annotationStore.selectAnnotation(c.id);
+        clickHandler();
+      });
+    }, [entity]);
+    const deleteAnnotation = useCallback(() => {
+      clickHandler();
+      confirm({
+        title: 'Delete annotation?',
+        body: (
+          <>
+            This will <strong>delete all existing regions</strong>. Are you sure you want to delete them?<br/>
+            This action cannot be undone.
+          </>
+        ),
+        buttonLook: 'destructive',
+        okText: 'Delete',
+        onOk: () => {
+          entity.list.deleteAnnotation(entity);
+        },
+      });
+    }, []);
+  
+    const isPrediction = entity.type === 'prediction';
+    const canSetAsTruth = isDefined(entity.pk);
+    const showGroundTruth = capabilities.groundTruthEnabled && !isPrediction && canSetAsTruth;
+
+    return (
+      <Block name="AnnotationButtonContextMenu">
+        {showGroundTruth && (
+          <Elem name='option' mod={{ groundTruth: true }} onClick={setGroundTruth}>
+            {isGroundTruth ? (
+              <>
+                <LsStar color='#FFC53D' width={iconSize} height={iconSize} /> Unset
+              </>
+            ) : (
+              <>
+                <LsStarOutline width={iconSize} height={iconSize} /> Set
+              </>
+            )} as Ground Truth
+          </Elem>
+        )}
+        <Elem name='option' mod={{ duplicate: true }} onClick={duplicateAnnotation}>
```

Comment:
@Travis1282 This button should only be enabled if the annotation is not a draft.